users/service.go

```diff
@@ -84,6 +84,9 @@ func (svc service) RegisterClient(ctx context.Context, token string, cli mgclien
 	}
 
 	if cli.Credentials.Secret != "" {
+		if !svc.passRegex.MatchString(cli.Credentials.Secret) {
```

Comment:
Let's move this to API layer (validate request) and use `apiutil.ErrValidation`.

Reply:
It is repeated both in API layer and service layer

Reply:
Let's group validation in the API layer.